src/js/components/stock-movement-wizard/outbound/AddItemsPage.jsx

```diff
@@ -676,12 +695,33 @@ class AddItemsPage extends Component {
             lineItems,
             val => ({
               ...val,
+              disabled: true,
               product: {
                 ...val.product,
                 label: `${val.productCode} ${val.product.translatedName ?? val.product.name}`,
               },
             }),
           );
+          // In autosave, we don't modify the state, because
+          // lines which have not passed the validation will be
+          // deleted during users work
+          if (!withStateChange) {
+            // We want to disable saved line, so I am looking for product with the same
+            // code and quantity higher than 0 in response
+            // (to avoid disabling new line with the same productCode)
+            const savedItemsProductCodes = lineItemsBackendData.map(item => item.productCode);
+            const lineItemsAfterSave = this.state.values.lineItems.map((item) => {
```

Comment:
Technically there is option to enter the same product twice (or even more). You should first look into `lineItem.id` in case the specific row was updated, and then if there is no `item.id` you can look into `product.id` or `product.productCode`